File: Programming Fundamentals/C#/Fundamentals/08.Text Processing/Exercise/P03_ExtractFile/P03_ExtractFile.cs
namespace P03_ExtractFile
{
    using System;
    using System.Linq;

    class P03_ExtractFile
    {
        static void Main(string[] args)
        {
            string path = Console.ReadLine();

            string file = path.Substring(path.LastIndexOf('\\') + 1);
            string fileExtension = file.Substring(file.IndexOf('.') + 1);
            string fileName = file.Substring(0, file.Length - fileExtension.Length - 1);

            Console.WriteLine($"File name: {fileName}");
            Console.WriteLine($"File extension: {fileExtension}");
        }
    }
}

File: Programming Fundamentals/JS/08.Text Processing/Exercise/p03_ExtractFile.js
function solve(path) {
  let file = path.substring(path.lastIndexOf('\\') + 1);

  let fileName = file.substring(0, file.lastIndexOf('.'));
  let extension = file.substring(file.lastIndexOf('.') + 1);

  console.log(`File name: ${fileName}`);
  console.log(`File extension: ${extension}`);
}

// Don't copy the calling of the function in judge, you won't get any points, just the code above
solve('C:\\Internal\\training-internal\\Template.pptx');
solve('C:\\Projects\\Data-Structures\\LinkedList.cs');
